cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -362,14 +392,25 @@ struct copy_intersecting_nbrs_and_update_intersection_size_t {
         thrust::tie(indices0, local_edge_offset0, local_degree0) =
           edge_partition.local_edges(edge_partition.major_offset_from_major_nocheck(major));
       }
+
+      if constexpr (!std::is_same_v<edge_property_value_t, thrust::nullopt_t>) {
+        property0 = edge_partition_e_value_input.value_first() + local_edge_offset0;
+      }
+
     } else {
       auto idx = first_element_to_idx_map.find(thrust::get<0>(pair));
       local_degree0 =
         static_cast<edge_t>(first_element_offsets[idx + 1] - first_element_offsets[idx]);
-      indices0 = first_element_indices.begin() + first_element_offsets[idx];
+      indices0           = first_element_indices.begin() + first_element_offsets[idx];
+      local_edge_offset0 = first_element_offsets[idx];
```

Comment:
Better move this before the line above and use this to update both indices0 and propety0.